packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -714,13 +714,17 @@ export class CollaborationWindow {
 	// reference segment for this client
 	currentSeq = 0;
 
+	// Grouped batching means seqNum can potentially be the same for different ops. Use clientSeqNum to differentiate
+	currentClientSeq? = 0;
+
 	localSeq = 0;
```

Comment:
I think `localSeq` is equivalent `clientSequenceNumber`? Or is `currentClientSeq` distinct from `clientSequenceNumber`?

Reply:
`currentClientSeq` is effectively `clientSequenceNumber. Does `localSeq` represent `clientSequenceNumber`, or is it just some local counter that is kept? How would sequence load compare `ISequenceDocumentMessage.clientSequenceNumber` to `localSeq`?

Reply:
`localSeq` is just a locally-kept counter, e.g. https://github.com/microsoft/FluidFramework/blob/912f6b4b94dfaee06f75df2a864dc76014821360/packages/dds/merge-tree/src/mergeTree.ts#L1712

I'm not sure how it compares to `ISequenceDocumentMessage.clientSequenceNumber` -- do the tests still pass if you use `collabWindow.localSeq` rather than `collabWindow.currentClientSeq`?

Reply:
I think using `localSeq` could result in wonky behavior if compared to `ISequenceDocumentMessage.clientSequenceNumber` as the `clientSequenceNumber` is not strictly sequential across multiple grouped batches.

For example:
Op 1, seqNum 1, clientSeqNum 1 -> localSeq is now 1
Op 2, seqNum 1, clientSeqNum 2 -> localSeq is now 2
Op 3, seqNum 2, clientSeqNum 1 -> localSeq is now 3
Op 4, seqNum 2, clientSeqNum 2 -> would the check fail? currentSeq == 2 and clientSeqNum (2) < localSeq (3)

Reply:
Why does `clientSequenceNumber` reset across batches?

Reply:
Grouped batching hijacks `clientSequenceNumber` to order ops in a batch (since ops in a grouped batch will all have the same `sequenceNumber`).
Without grouped batching, the `clientSequenceNumber` would incrementally increase per client. But there could be many clients sending ops one after the other for this sequence and also ops coming from different DDSs.

Reply:
I tried replacing the check in `SharedSegmentSequence.loadCore(...)` with `m.clientSequenceNumber <= collabWindow.localSeq` and the test doesn't fail, but the `collabWindow.localSeq` doesn't stays at 0.
The increment only happens if a seqNum is unassigned, which shouldn't be the case for these ops as they were already sent across the wire.

Reply:
For visibility: chatted with Connor about using `localSeq` and I don't think it will work for this case